web-common/src/features/canvas/components/BaseCanvasComponent.ts

```diff
@@ -68,11 +77,95 @@ export abstract class BaseCanvasComponent<T = ComponentSpec> {
 
     this.resource.set(resource);
     this.id = resource.meta?.name?.name as string;
-    this.state = new CanvasComponentState(
+
+    const timeFiltersStore: SearchParamsStore = (() => {
+      const store = derived(this.specStore, (spec) => {
+        return new URLSearchParams(spec?.["time_filters"] ?? "");
+      });
+      return {
+        subscribe: store.subscribe,
+        set: (key: string, value: string | undefined) => {
+          const searchParams = get(store);
+
+          if (value === undefined || value === null || value === "") {
+            searchParams.delete(key);
+          } else {
+            searchParams.set(key, value);
+          }
+
+          this.updateProperty(
+            "time_filters" as AllKeys<T>,
+            searchParams.toString() as T[AllKeys<T>],
+          );
+          return true;
+        },
+        clearAll: () => {
+          const searchParams = get(store);
+
+          searchParams.forEach((_, key) => {
+            searchParams.delete(key);
+          });
+
+          this.updateProperty(
+            "time_filters" as AllKeys<T>,
+            searchParams.toString() as T[AllKeys<T>],
+          );
+        },
+      };
+    })();
+
+    const localFiltersStore: SearchParamsStore = (() => {
```

Comment:
Is this supposed to be `localFiltersStore` ? An object for `localFilters` has already been defined below. The terminology is a bit confusing between `timeFiltersStore`, `localFiltersStore`, `localFilters` and `localTimeControls`.